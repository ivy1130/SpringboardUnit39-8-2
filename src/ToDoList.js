import React, { useState } from "react";
import { v4 as uuid } from 'uuid';
import ToDo from "./ToDo"
import NewToDoForm from "./NewToDoForm";

const ToDoList = () => {
  const INITIAL_STATE=[]

  const [toDos, setToDos] = useState(INITIAL_STATE)
  const addToDo = (newToDo) => {
    setToDos(toDos => [...toDos, {...newToDo, id: uuid()}])
  }

  const removeToDo = (id) => {
    setToDos(toDos.filter(toDo => toDo.id !== id))
  }

  return (
    <div className="ToDoList">
      <h3>To Do List</h3>
      <NewToDoForm addToDo={addToDo} />
      <div>
        {toDos.map(({ task, id }) => 
          <ToDo task={task} id={id} key={id} removeToDo={removeToDo}/>
        )}
      </div>
    </div>
  )

}

export default ToDoList