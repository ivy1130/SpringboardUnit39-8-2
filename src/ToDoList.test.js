import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ToDoList from "./ToDoList";

function addToDo(toDoList, task="write rtl tests") {
  const taskInput = toDoList.getByLabelText("Task:");
  fireEvent.change(taskInput, { target: { value: task } });
  const button = toDoList.getByText("Add Task!");
  fireEvent.click(button);
}

it("renders without crashing", function() {
  render(<ToDoList />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<ToDoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a new task", function() {
  const toDoList = render(<ToDoList />);

  // no boxes yet
  expect(toDoList.queryByText("x")).not.toBeInTheDocument();

  addToDo(toDoList);

  // expect to see a box
  const removeButton = toDoList.getByText("x");
  expect(removeButton).toBeInTheDocument();
  expect(removeButton.parentElement).toHaveTextContent("write rtl tests");
  // expect form to be empty
  expect(toDoList.getAllByDisplayValue("")).toHaveLength(1);

  // expect(asFragment()).toMatchSnapshot();
});

it("can remove a to do", function() {
  const toDoList = render(<ToDoList />);
  addToDo(toDoList);

  const removeButton = toDoList.getByText("x");

  // click the remove button and the box should be gone
  fireEvent.click(removeButton);
  expect(removeButton).not.toBeInTheDocument();
});
