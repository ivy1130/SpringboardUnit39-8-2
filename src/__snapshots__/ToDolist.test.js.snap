// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ToDoList"
  >
    <h3>
      To Do List
    </h3>
    <form>
      <label
        for="task"
      >
        Task:
      </label>
      <input
        id="task"
        name="task"
        placeholder="task"
        type="text"
        value=""
      />
      <button>
        Add Task!
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
